src/pods/canvas/components/table/database-table.component.tsx

```diff
@@ -101,120 +97,38 @@ export const DatabaseTable: React.FC<Props> = ({
-  };
-
-  const handleClick = (e: React.MouseEvent<SVGGElement, MouseEvent>) => {
+  const handleSelectTable = () => {
     if (!isSelected) {
       selectTable(tableInfo.id);
     }
-    e.stopPropagation();
   };
 
   const handleDoubleClick = () => {
     onEditTable(tableInfo);
   };
-  const pencilIconClick = (e: React.MouseEvent<SVGGElement, MouseEvent>) => {
-    onEditTable(tableInfo);
-    e.stopPropagation();
-  };
 
   return (
     <g
       transform={`translate(${tableInfo.x}, ${tableInfo.y})`}
       onMouseDown={onMouseDown}
       className={classes.tableContainer}
     >
-      <defs>
-        <filter id="table_component_selected" x="0" y="0">
-          <feDropShadow
-            dx="3"
-            dy="3"
-            stdDeviation="4"
-            floodColor="var(--shadow-filter)"
-          />
-        </filter>
-      </defs>
-      <rect
-        x="0"
-        y="0"
-        rx={TABLE_BORDER_RADIUS}
-        width={TABLE_CONST.TABLE_WIDTH}
-        height={totalHeight + TABLE_CONST.HEADER_TITLE_GAP}
-        className={classes.tableBackground}
-        style={rectStyle}
-        stroke="1.2"
+      <DatabaseTableBorder totalHeight={totalHeight} isSelected={isSelected} />
+      <DatabaseTableHeader
+        onEditTable={handleDoubleClick}
+        onSelectTable={handleSelectTable}
+        isSelected={isSelected}
+        tableName={tableInfo.tableName}
       />
-      <rect
-        x="0"
-        y={TABLE_CONST.HEADER_HEIGHT - 5}
-        width={TABLE_CONST.TABLE_WIDTH}
-        height="5"
-        className={classes.tableHeader}
-      />
-      <rect
-        rx={TABLE_BORDER_RADIUS}
-        x="0"
-        y="0"
-        width={TABLE_CONST.TABLE_WIDTH}
-        height={TABLE_CONST.HEADER_HEIGHT}
-        className={classes.tableHeader}
-        onClick={e => {
-          e.stopPropagation();
-        }}
-        onDoubleClick={handleDoubleClick}
-      />
-      <TruncatedText
-        text={tableInfo.tableName}
-        x={TITLE_MARGIN_LEFT}
-        y={4}
-        width={TABLE_CONST.TABLE_WIDTH - TITLE_MARGIN_LEFT}
-        height={TABLE_CONST.FONT_SIZE}
-        textClass={classes.tableText}
-      />
-      {isSelected && (
-        <g
-          transform={`translate(${TABLE_CONST.TABLE_WIDTH - (PENCIL_ICON_WIDTH - PENCIL_MARGIN_RIGHT)}, 2)`}
-          onClick={pencilIconClick}
-        >
-          <rect
-            x="0"
-            y="0"
-            width={PENCIL_ICON_WIDTH + PENCIL_MARGIN_RIGHT}
-            fill="transparent"
-            height="25"
-            onClick={pencilIconClick}
-            style={{ cursor: 'pointer' }}
-          />
-          <Edit />
-        </g>
-      )}
-      ;
-      <g transform={`translate(0, ${TABLE_CONST.HEADER_TITLE_GAP})`}>
-        {renderedRows}
-      </g>
-      <rect
+      <DatabaseTableBody renderedRows={renderedRows} />
+      {/* TODO: Check with the team, is this rect neccessary? */}
```

Comment:
I have remove the rect below and everything seems to be working fine, anybody know what is this rect for?